code/datums/mood_events/mood_event.dm

```diff
@@ -4,18 +4,67 @@
 	var/timeout = 0
 	var/hidden = FALSE//Not shown on examine
 	var/category //string of what category this mood was added in as
+	var/special_screen_obj //if it isn't null, it will replace or add onto the mood icon with this. see happiness drug for example
+	var/special_screen_replace = TRUE //if false, it will be an overlay instead
 	var/mob/owner
 
 /datum/mood_event/New(mob/M, param)
 	owner = M
 	add_effects(param)
+	if(special_screen_obj)
+		add_icon()
 
 /datum/mood_event/Destroy()
 	remove_effects()
+	if(special_screen_obj)
+		remove_icon()
 	return ..()
 
 /datum/mood_event/proc/add_effects(param)
 	return
 
+/datum/mood_event/proc/add_icon() //does not replace the icon
+	GET_COMPONENT_FROM(mood, /datum/component/mood, owner)
+	if(special_screen_replace)//if it's replacing, lets make sure we're not replacing a more important (higher mood change) icon
+		var/highest_absolute_mood = 0
+		for(var/i in mood.mood_events)
+			var/datum/mood_event/event = i
+			if(!event.special_screen_obj || !special_screen_replace || event == src)

```

Comment:
Is this supposed to be !event.special_screen_replace? Because you already tested src.special_screen_replace above

Reply:
you're right, i missed that my bad. this will probably look a lot better on the component